templates/definition/meter/goodwe-hybrid.yaml

```diff
@@ -92,5 +92,56 @@ render: |
       type: holding
       decode: uint32
     scale: 0.1
+  batterymode:
+    source: watchdog
+    timeout: 30s
+    reset: 1 # reset watchdog on normal
+    set:
+      source: switch
+      switch:
+      - case: 1 # normal
+        set:
+          source: const
+          value: 1 # Normal operation mode
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 47511 # EMSPowerMode
+              type: writesingle
+              encoding: uint16
+      - case: 2 # hold
+        set:
+          source: const
+          value: 8 # Battery standby
+          set: 
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 47511 # EMSPowerMode
+              type: writesingle
+              encoding: uint16
+      - case: 3 # charge
+        set:
+          source: sequence
+          set:
+          - source: const
+            value: 2 # Charge from PV+AC according to the EMSPowerSet
+            set: 
+              source: modbus
+              {{- include "modbus" . | indent 12 }}
+              register:
+                address: 47511 # EMSPowerMode
+                type: writesingle
+                encoding: uint16
+          - source: const
+            value: 2500 # W
```

Comment:
Wo kommt dieser Wert her?

Reply:
In dem Eingangs erwähnten Diskussions-thread hab ich das auch gefragt, da der Wert auch so bei HUAWEI Batteriesperre verwendet wird. 
Meines Erachtens ist es schwer den Wert konstant zu definieren, da die Ladeleistung von sehr vielen Parametern abhängig ist. Von daher ist die Festlegung der Konstante ein offener Punkt fuer alle  Batteriesperren fuer alle Wechselrichter.
Mein Vorschlag (aber imho nicht scope dieses PR):
Den Default-Wert maximal halten (z.b.: 10000 oder mehr). Der Wechselrichter wird das schon selber auf den wirklich möglichen max Wert laufen lassen.  
Und zusätzlich fuer diesen Charge Mode den Nutzer in der GUI selber einstellen lassen wieviel Leistung er denn vom Netz maximal nehmen lassen will. Diesen Wert sollte dann aber jeder im template der jeweiligen Wechselrichters übernehmen.

Bin aber natürlich offen fuer andere Vorschläge.

Reply:
Die Frage ist was ist hier der Default und gilt der für alle Varianten?

Reply:
Bei mir auf dem Wechselrichter war der Wert "0".
Hab leider keine offizielle Dokumentation und hab auch keine im Internet gefunden.
Folgendes findet sich bei OpenEMS [Link](https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.goodwe/src/io/openems/edge/goodwe/common/enums/EmsPowerMode.java) GitHub als Erklärung zu den Modus 2 (Charge) :
Scenario: Control the battery to keep charging.
	
	 * <p>
	 * PBattery = Xmax + PV (Charge)
	 *
	 * <p>
	 * Xmax is to allow the power to be taken from the grid, and PV power is
	 * preferred. When set to 0, only PV power is used. Charging power will be
	 * limited by charging current limit.
Xmax ist in dem Fall das Register 47512.

Sprich im Wechselrichter war der Default-wert "0" also PV only. 
Aber so wie ich das verstanden habe sollte der Charge modus ja Leistung vom Netz nehmen.
Und da hab einfach erstmal den Wert wie bei HUAWEI genommen, 2500W.
Damit darf der Wechselrichter von Netz maximal 2500 W nehmen um damit die Batterie zu laden.

Hier ein Beispiel wie das aussieht mit dem Befehl von dir unten.
evcc meter -b charge 

![Screenshot 2024-03-10 at 16 06 33](https://github.com/evcc-io/evcc/assets/138592105/f8eaa510-8c9b-43c5-8d16-887f7817a615)

Reply:
Was hat Huawei mit GoodWe zu tun?

Was ist denn der Maximalwert den man da angeben kann?

Reply:
Der Zusammenhang von Huawei und GoodWe wäre eigentlich in dem Thread #12752 erkennbar. Kurz gesagt, ich nehm einfach den gleichen Wert wie Huawei fuer max. Power.

GoodWe Maximalwert: Wie gesagt, ich habe keine offizielle Dokumentation. Bei meinem Wechselrichter konnte ich erfolgreich Werte bis 20000 W setzen. Alles darueber meckerte NodeRed beim setzen.

Aber um die Diskussion zu beschleunigen, wenn es Anforderungen zu diesen Charge Modus bezüglich den Max-Wert vom Grid gibt versuche ich die gerne umzusetzen. Momentan kenn ich sie nicht.

Reply:
Wie kommen wir hier weiter?
Ich denke mit einem Konstanten Wert, egal wie wir den wählen, werden wir keine finale Lösung finden.
Mein Vorschlag, wie anfangs erwähnt, diesen PR erstmal mit diesen temporären Wert (2500 oder auch anderen Wert) übernehmen. (Ist ja auch so bei anderen Wechselrichter gemacht)
Parallel einen Issue aufmachen um eine sinnvolle Festlegung des Wertes fuer alle Wechselrichter zu definieren.

Reply:
Sorry @andiwist mit GoodWe kann ich da auch nicht weiterhelfen. Bei Huawei ist es schlicht die Natur der Sache, dass JEDER Luna Speicher eine Ladeleistung von 2500W annimmt. Daher ist das an dieser Stelle kein Problem.

Bei Huawei könnte man allerdings sogar die vom Nutzer eingestellte Ladegrenze auslesen, in eine Variable Zwischenspeichern und anschließend für den Mode "charge" verwenden. Hatte ich mal vorgeschlagen, aber meine Programmierkenntnisse reichen bei weitem nicht dafür aus hier selber einen PR zu erstellen :-(

Reply:
Ja, ich kenne auch HUAWEI und da hat ein Modul max. Ladeleistung 2.5kw richtig. Kommt noch Modul on top, kann es dann 5kw. Vermutlich gehen auch mehr. 2,5kw ist quasi der kleinste gemeinsame Nenner.
Ähnlich auch bei GoodWe, aber da ist es noch komplizierte, da nicht nur GoodWe Speicher sondern auch andere Hersteller angeschlossen werden koennen, die wiederum verschiedene Max Ladeströme haben. 

Also, ich glaub momentan führt kein Weg vorbei, einfach maximal möglichen Wert einstellen, 20000 W. Hab sonst keine Lösung.

Reply:
Genau, dann 10000.